lib/livebook_web/live/session_live/export_component.ex

```diff
@@ -7,17 +7,11 @@ defmodule LivebookWeb.SessionLive.ExportComponent do
   def update(assigns, socket) do
     socket = assign(socket, assigns)
 
-    socket =
-      if socket.assigns[:notebook] do
-        socket
-      else
-        # Note: we need to load the notebook, because the local data
-        # has cell contents stripped out
-        notebook = Session.get_notebook(socket.assigns.session.pid)
-        assign(socket, :notebook, notebook)
-      end
-
-    {:ok, socket}
+    {:ok,
+     socket
+     # Note: we need to load the notebook, because the local data
+     # has cell contents stripped out
+     |> assign_new(:notebook, fn -> Session.get_notebook(socket.assigns.session.pid) end)}
   end
```

Comment:
```suggestion
    |> assign_new(:any_stale_cell?, any_stale_cell?)
  end
```

Reply:
How about this?
```elixir
|> assign_new(:any_stale_cell?, fn -> Map.pop!(assigns, :any_stale_cell?) end)
```

Reply:
We need to pop it before we do `socket = assign(socket, assigns)`, otherwise it's going to be already assigned at this point :)

Reply:
Oh I got it!